core/src/main/scala/org/http4s/headers/Forwarded.scala

```diff
@@ -80,14 +84,58 @@ object Forwarded
     sealed abstract case class Obfuscated private (value: String) extends Name with Port
 
     object Obfuscated {
+      val parser: Parser1[Obfuscated] =
+        (P.char('_') ~
+          P.rep1(P.oneOf1(List(Rfc5234.alpha, Rfc5234.digit, P.charIn("._-"))), 1)).string
+          .map(Obfuscated.apply)
+
       def fromString(s: String): ParseResult[Obfuscated] =
-        new ModelNodeObfuscatedParser(s).parse
+        parser.parseAll(s).left.map { e =>
+          ParseFailure(s"invalid obfuscated value '$s'", e.toString)
+        }
 
       /** Unsafe constructor for internal use only. */
       private[http4s] def apply(s: String): Obfuscated = new Obfuscated(s) {}
     }
 
-    def fromString(s: String): ParseResult[Node] = new ModelNodeParser(s).parse
+    def fromString(s: String): ParseResult[Node] =
+      parser.parseAll(s).left.map { e =>
+        ParseFailure(s"invalid node '$s'", e.toString)
+      }
+
+    val parser: Parser1[Node] = {
+      // https://tools.ietf.org/html/rfc7239#section-4
+
+      def modelNodePortFromString(str: String): Option[Node.Port] =
+        Try(Integer.parseUnsignedInt(str)).toOption.flatMap(Node.Port.fromInt(_).toOption)
+
+      // node-port = port / obfport
+      // port      = 1*5DIGIT
+      // obfport   = "_" 1*(ALPHA / DIGIT / "." / "_" / "-")
+      val nodePort: P[Node.Port] =
+        Numbers.digits1
+          // is it worth it to consume only up to 5 chars or just let it fail later?
```

Comment:
A `max` is coming in cats-parse-0.3 that will make this easier.  We can let it fail later and clean it up on upgrade, I'd say.